code/modules/atmospherics/machinery/components/binary_devices/thermomachine.dm

```diff
@@ -178,6 +194,18 @@
 			update_appearance()
 			update_parents()
 			return
+		if(enviroment.temperature > FUSION_MAXIMUM_TEMPERATURE && safeties)
+			on = FALSE
+			visible_message("<span class='warning'>The enviroment's temperature has reached critical levels, shutting down...</span>")
+			update_appearance()
+			return
+		else if(enviroment.temperature > FUSION_MAXIMUM_TEMPERATURE && !safeties)
+			if((REALTIMEOFDAY - lastwarning) / 5 >= WARNING_DELAY)
+				lastwarning = REALTIMEOFDAY
+				visible_message("<span class='warning'>The enviroment's temperature has reached critical levels!</span>")
+				if(check_explosion(enviroment.temperature))
+					explode()
+					return PROCESS_KILL //we dying anyway, so let's stop processing
```

Comment:
I really wish this code was better organized, but we can deal with that later

Reply:
yeah but i don't know how to